faculty/clients/environment.py

```diff
@@ -42,16 +186,77 @@ class EnvironmentSchema(BaseSchema):
     author_id = fields.UUID(data_key="authorId", required=True)
     created_at = fields.DateTime(data_key="createdAt", required=True)
     updated_at = fields.DateTime(data_key="updatedAt", required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
+    def make_environment(self, data):
+        return Environment(**data)
+
+
+class EnvironmentUpdateSchema(BaseSchema):
+    name = fields.String(required=True)
+    description = fields.String(required=True)
+    specification = fields.Nested(SpecificationSchema(), required=True)
+
+    @post_load
```

Comment:
AFAICT this schema is only used to `dump` - this `post_load` method is never used so should be removed.

On a similar note, `required` is only respected on `load` and so we should probably be removed from the fields above as it's misleading.

Reply:
In the future I intend to use the client to load environment updates from, for example, JSON files. So I believe there is a use case for loading environment updates.